core/coreunix/add.go

```diff
@@ -200,21 +207,35 @@ func (adder *Adder) Finalize() (*dag.Node, error) {
 		return nil, err
 	}
 
-	return root, nil
+	rootNode, err = root.GetNode()
+	if err != nil {
+		return nil, err
+	}
+	return rootNode, nil
 }
 
-func (adder *Adder) outputDirs(path string, nd *dag.Node) error {
-	if !bytes.Equal(nd.Data, folderData) {
+func (adder *Adder) outputDirs(path string, fs mfs.FSNode) error {
+	nd, err := fs.GetNode()
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(nd.Data, folderData) || fs.Type() != mfs.TDir {
```

Comment:
I noticed that tests break when I remove it -- but only the trickledag ones. Perhaps the root node created isn't being set to a `TDir`?